### DISCUSSION TITLE
Any workaround for { where: {field: undefined} } returning a random row?

### DISCUSSION THREAD

**Discussion by kasraghoreyshi** [prisma/prisma#12772] [Q&A] [answered]
So I guess this is the expected behavior where if you give undefined to the where query, it will return a random result to you instead of returning undefined back. For example running something like this will cause prisma to return a random row instead of undefined:
```js
this.prismaService.user.findFirst({
      where: { email: undefined },
    });
```
 Is there any workaround for this? I know that validation will most likely prevent anything catastrophic from happening but It's a bit worrying to think that if someone manages to send an undefined field in an endpoint like delete account, it could possibly delete someone else's account. Right now I'm just manually checking if the input is not undefined but I want a more robust solution if there is any. Thanks

**Comment by austincrim** [comment, marked as answer]
Hey @kasraghoreyshi 👋 ,

The Prisma methods that require a row to be uniquely identified will throw an error if the `where` condition receives `undefined` or `null`. `findFirst` and `findMany` don't require that you pass anything to the `where` clause, so we don't error in that case.

If you want a centralized spot to put query validation, I would look into [Prisma middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware).

Let me know if that helps!

**Reply by kasraghoreyshi** [reply]
I honestly didn't know anything about prisma middlewares and they seem extremely useful. Thanks, I really appreciate your answer!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
